bloom-housing/bloom
# feat: skip preferences if not on listing

# Pull Request Template

## Issue Overview

This PR addresses #2124


## Description

This updates `retrieveApplicationConfig` to accept a listing and not include preferences if they don't exist on the listing. This also updates fetching listings for the application to only fetch the listing a user is applying to, instead of all of them and then filtering them out.

## How Can This Be Tested/Reviewed?

Apply to a listing that has preferences and you should see preferences in the progress bar as well as in the form. Apply to a listing that has zero preferences and you should no longer see preferences in the progress bar or in the form.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
